xayn-ai/src/coi/system.rs

```diff
@@ -866,101 +960,203 @@ mod tests {
     }
 
     #[test]
-    fn test_compute_weights_empty_cois() {
+    fn test_compute_relevances_empty_cois() {
+        let config = Configuration {
+            horizon: Duration::from_secs_f32(SECONDS_PER_DAY),
+            ..Configuration::default()
+        };
+        let mut system = CoiSystem::new(config, SMBert::default());
         let cois = create_pos_cois(&[[]]);
-        let horizon = Duration::from_secs_f32(SECONDS_PER_DAY);
-        let weights = CoiSystem::default().compute_weights(&cois, horizon);
-        assert!(weights.is_empty());
+        system.compute_relevances(&cois);
+        assert!(system.coi_to_relevance.is_empty());
+        assert!(system.relevance_to_key_phrase.is_empty());
     }
 
     #[test]
-    fn test_compute_weights_zero_horizon() {
+    fn test_compute_relevances_zero_horizon() {
+        let config = Configuration {
+            horizon: Duration::ZERO,
+            ..Configuration::default()
+        };
+        let mut system = CoiSystem::new(config, SMBert::default());
         let cois = create_pos_cois(&[[1., 2., 3.], [4., 5., 6.]]);
-        let horizon = Duration::ZERO;
-        let weights = CoiSystem::default().compute_weights(&cois, horizon);
-        assert_approx_eq!(f32, weights, [0., 0.]);
+        system.compute_relevances(&cois);
+        assert_eq!(system.coi_to_relevance.len(), 2);
+        assert_approx_eq!(f32, system.coi_to_relevance[&cois[0].id], [0.]);
+        assert_approx_eq!(f32, system.coi_to_relevance[&cois[1].id], [0.]);
+        assert!(system.relevance_to_key_phrase.is_empty());
+    }
+
+    fn dedup<const N: usize>(array: [f32; N]) -> Array1<f32> {
+        assert!(IntoIterator::into_iter(array).all(f32::is_finite));
+        IntoIterator::into_iter(array)
+            .map(Relevance)
+            .collect::<BTreeSet<_>>()
```

Comment:
Can't we use an hashset? Or we also want them ordered?

Reply:
correct, we want them ordered to compare in the tests without additional sorting.